web/ui/react-app/src/components/CustomInfiniteScroll.tsx

```diff
@@ -0,0 +1,50 @@
+import { ComponentType, useEffect, useState } from 'react';
+import InfiniteScroll from 'react-infinite-scroll-component';
+
+const initialNumberOfTargetsDisplayed = 50;
+
+export interface InfiniteScrollItemsProps<T> {
+  items: T[];
+}
+
+interface InfiniteScrollProps<T> {
+  value: T[];
+  child: ComponentType<InfiniteScrollItemsProps<T>>;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+const CustomInfiniteScroll = <T extends unknown>({ value, child }: InfiniteScrollProps<T>) => {
```

Comment:
I'd name `value` to something like `values` or maybe even better `allItems`?

Reply:
I chose `allItems` I think it makes more sense with the name of the component